### ISSUE TITLE
Document update fails with php-mongo 1.5.1 + genghis 2.3.11

### ISSUE THREAD

**Issue by willianpts** [bobthecow/genghis#172]
Document update fails with "Error updating document."
Some system info:

```
$ext = new ReflectionExtension('mongo');
echo $ext->getVersion(); // outputs 1.5.1
echo phpversion(); // outputs 5.5.11
```

Here's the apache error:

```
[core:notice] [pid 9105] AH00052: child pid 9109 exit signal Segmentation fault (11)
```

php-mongo 1.4.5 dont have the issue.

**Comment by bobthecow** [comment]
I'll take a look, but in the meantime can you check a few more things? Does it fail on inserts? Does it fail on every update, or just specific ones?

**Comment by willianpts** [comment]
- Insert works correctly;
- Any document update fails, be it a field insertion, field removal or field edit;
- Document update fails at any database/collection;

**Comment by willianpts** [comment]
With the latest update of the php extension the bug disappeared.
http://pecl.php.net/package/mongo/1.5.2

php-mongo 1.4.5: no issue;
php-mongo 1.5.1: update fails as described;
php-mongo 1.5.2: no issue;

hmm, maybe you can simply blacklist and forget about 1.5.1

**Comment by bobthecow** [comment]
Yeah, must have been a regression in 1.5.1. Blacklisting it sounds like a great answer :)
